### PR TITLE
[Form] Remove hard dependency on symfony/intl

### LINKED ISSUE #29229: [Form] Move symfony/intl to require-dev
In #28610 we've added exceptions to give better error messages in case one uses a form type that relies on symfony/intl. The problem is this can never happen, as sf/intl is a hard requirement per composer.json

https://github.com/symfony/symfony/blob/dbf053bc854f6768ddcd8ed39f7cbb2c21e500e6/src/Symfony/Component/Form/composer.json#L18-L21

This compared to the Validator component, where it's a soft requirement.

Is there any hassle-free approach for 4.x or 5.0 here? (in an effort to move it to require-dev instead).

### LINKED ISSUE #39596: [Form] Remove dependency on symfony/intl
The form component have a hard dependency on `symfony/intl`. I like that component, but it feels a bit overkill to download 18MB of code for if you only are creating a small form. 

Currently, there is only the LocaleType, CountryType, CurrenyType, LanguageType and TimezoneType that is using `symfony/intl`. I suggest that we should remove Form component's dependency in Intl in Symfony 6 to make the form component lighter.

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 5.x (or 6.0)
| Bug fix?      | 
| New feature?  | no
| Deprecations? | yes
| Tickets       | Fix #39596
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/15026

This was voted down in 2018 (#29229) and will revert #29720 by @chalasr. I reopen it because the Form component is way less dependent on Intl component now.  

Im hesitant if we should do this in 5.x or 6.0. If a user don't have `symfony/intl` installed, they will get an error in runtime. That is something that speaks for doing it in 6.0. 

Could I get some opinions?
 
### TODO

- [x] Update `UPGRADE-x.x.md`

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php [hunk 1]
```diff
@@ -27,6 +29,10 @@ public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults([
             'choice_loader' => function (Options $options) {
+                if (!class_exists(Intl::class)) {
+                    throw new LogicException(sprintf('The "symfony/intl" component is required to use "%s".', static::class));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php]: to all messages: `Try running "composer require symfony/intl".`

**Review by nicolas-grekas — COMMENTED** [review]
I'm fine doing it in 5.x personally.
Of course, this should be documented in CHANGELOG+UPGRADE files, in the doc, ~and the website-skeleton also should be updated~ (it has explicit "intl" as a dep since 2018).

File: src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php (hunk 1 above)
Comment (Nyholm) [reply src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php]: Thank you

**Comment by Nyholm** [comment]
The PR is updated and documentation is added.

File: src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php [hunk 2]
```diff
@@ -47,6 +48,10 @@ public function configureOptions(OptionsResolver $resolver)
                 $input = $options['input'];
 
                 if ($options['intl']) {
+                    if (!class_exists(Intl::class)) {
+                        throw new LogicException(sprintf('The "symfony/intl" component is required to use "%s" with option "intl=true". Try running "composer require symfony/intl".', static::class));
```
Comment (Nyholm) [inline src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php:52]: Note that this exception message is different.

**Review by nicolas-grekas — APPROVED** [review]
symfony/intl is listed in the website-skeleton since 2018

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Nyholm.

**Comment by Nyholm** [comment]
Thank you for merging

**Pull request merged**